templates/tfengine/components/cicd/variables.tf

```diff
@@ -12,10 +12,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+variable "build_editors" {
+  type = list(string)
+}
+
+variable "build_viewers" {
+  type = list(string)
+}
+
 variable "billing_account" {
   type = string
 }
 
+{{- if has . "cloud_source_repository"}}
+
+variable "cloud_source_repository" {
+  type = object({
+    name = string
+    {{- if has .cloud_source_repository "readers"}}
+    readers = list(string)
+    {{- end}}
+    {{- if has .cloud_source_repository "writers"}}
+    writers = list(string)
+    {{- end}}
+  })
+}
+{{- end}}
```

Comment:
What is the purpose of the ifs here? I think these variables can always be present but have a default of null. They can still only conditionally be rendered in main.tf.

Reply:
They're because the cloud_source_repository and the github variables are mutually exclusive. If they're declared, they'll be used on the resources even though they're null, and it cannot create the triggers.

Am I correct?

Reply:
I looked again and you're right, the TF resource only accepts one, not both, so we need to keep this templating.